pyrevolve/revolve_bot/revolve_module.py

```diff
@@ -265,6 +323,38 @@ def to_sdf(self, tree_depth='', parent_link=None, child_link=None):
         parent_link.append(imu_sensor)
         return visual, collision, imu_sensor
 
+    def count_module_connections(self) -> int:
+        """
+        Count how many connections the module has.
+        Connected TouchSensor and BrickSensor Modules are ignored
+        :return: number of connections
+        """
+        # The CoreModule does not have a parent module
+        return super(CoreModule, self).count_module_connections() - 1
+
+    def is_folding(self) -> bool:
+        # parent slot is not supposed to be set
+        parent_slot = None
+        target_slot = None
+        for slot, child in self.iter_children():
+            # is the child slot occupied?
+            if child is None:
+                continue
+            # ignore TouchSensor and BrickSensor modules
+            if isinstance(child, TouchSensorModule) or \
```

Comment:
You could inherit TouchSensorModule from SensorModule (create new class) and inherit BrickSensorModule from SensorModule so you only have to do one check instead of two.

Reply:
yeah, good point. But I'm not going to touch that right now. It's out of scope and I'm afraid I'm going to break stuff

Reply:
Could literaly do 

class SensorModule:
    pass

class TouchSensorModule(SensorModule):
   ....

class BrickSensorModule(SensorModule):
   ...

But yeah, that is up to you.